src/doc/doc-builders.js

```diff
@@ -1,5 +1,17 @@
 "use strict";
 
+/**
+ * @typedef {Object} DocObject
```

Comment:
This type should eventually become a tagged union.

Reply:
The DocObject is just one of the 2 options in the tagged union which I defined below. Can you think of a better way to do this?

Reply:
The union you defined is not tagged. We should define a separate type for each kind of Doc, and each subtype's field `type` will be of a string literal type. But it's totally okay as is for now. Let's do it later.

Reply:
> We should define a separate type for each kind of Doc, and each subtype's field `type` will be of a string literal type.

I would appreciate any pointers.

> Let's do it later.

Any ideas show we should avoid forgetting this?

Reply:
https://basarat.gitbooks.io/typescript/docs/types/discriminated-unions.html

A TBD comment will do.